Lib/pdb.py

```diff
@@ -414,16 +416,32 @@ def preloop(self):
                     self.message('display %s: %r  [old: %r]' %
                                  (expr, newvalue, oldvalue))
 
-    def interaction(self, frame, traceback):
+    def interaction(self, frame, tb_or_exc):
         # Restore the previous signal handler at the Pdb prompt.
+
+        _exceptions = []
+        if isinstance(tb_or_exc, BaseException):
+            tb_or_exc, exception = tb_or_exc.__traceback__, tb_or_exc
```

Comment:
I think it's easier to follow if we keep the name of the traceback variable as is, and do this:

```suggestion
            traceback, exception = tb_or_exc.__traceback__, tb_or_exc
```